_posts/2019-05-24-varying-slopes/varying_slopes.Rmd

```diff
@@ -0,0 +1,748 @@
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+
+In a previous post, we showed how to use [tfprobability](https://rstudio.github.io/tfprobability/) -- the R interface to TensorFlow Probability -- to build a _multilevel_, or  _partial pooling_, model of tadpole survival in differently sized (and thus, differing in inhabitant number) tanks.
+
+A completely _pooled_ model would have resulted in a global estimate of survival count, irrespective of tank; an _unpooled_ model would have learned to predict survival count for each tank separately. The former approach does not take into account different circumstances; the latter does not make use of common information. (Also, it clearly has no predictive use unless we want to make predictions for the very same entities we used to train the model.)
+
+In contrast, a _partially pooled_ model lets you make predictions for the familiar, as well as new entities: Just use the appropriate prior.
+
+Assuming we _are_ in fact interested in the same entities -- why would we want to apply partial pooling?
+For the same reasons so much effort in machine learning goes into devising regularization mechanisms.
+We don't want to overfit too much to actual measurements, be they related to the same entity or a class of entities. If I want to predict my heart rate as I wake up next morning, based on a single measurement I'm taking now (let's say it's evening and I'm frantically typing a blog post), I better take into account some facts about heart rate behavior in general (instead of just projecting into the future the exact value measured right now).
+
+In the tadpole example, this means we expect generalization to work better for tanks with many inhabitants, compared to more solitary environments. For the latter ones, we better take a peek at survival rates from other tanks, to supplement the sparse idiosyncratic information available. 
+Or using the technical term, in the latter case we hope for the model to _shrink_ its estimates toward the overall mean more noticeably than in the former.
+
+This type of information sharing is already very useful, but it gets better. The tadpole model is a _varying intercepts_ model, as McElreath calls it (or _random intercepts_, as it is sometimes -- confusingly -- called ^[cf. [the Wikipedia article on multilevel models](https://en.wikipedia.org/wiki/Multilevel_model) for a collection of terms encountered when dealing with this subject, and e.g. [Gelman's](https://statmodeling.stat.columbia.edu/2005/01/25/why_i_dont_use/) dissection of various ways _random effects_ are defined]) -- _intercepts_ referring to the way we make predictions for entities (here: tanks), with no predictor variables present. So if we can pool information about intercepts, why not pool information about _slopes_ as well? This will allow us to, in addition, make use of _relationships_ between variables learnt on different entities in the training set.
```

Comment:
> This will allow us to, in addition, make use of _relationships_ between variables learnt on different entities in the training set.

I'm not totally sure what "learnt" refers to hear (the variables examined in the different training sets?) If this is a common term that I just don't know (very possible), feel free to totally ignore.